packages/common/src/hooks/useDialog/useDialog.tsx

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogTitle from '@mui/material/DialogTitle';
+import { LocaleKey, useTranslation } from '../../intl/intlHelpers';
+import { ArrowRightIcon, CheckIcon, XCircleIcon } from '../../ui/icons';
+import { DialogButton } from '../../ui/components/buttons/DialogButton';
+
+export interface ButtonProps {
+  icon?: React.ReactElement;
+  labelKey?: LocaleKey;
+  onClick?: () => void;
+  visible?: boolean;
+}
+export interface DialogButtonOverrides {
+  cancel?: ButtonProps;
+  ok?: ButtonProps;
+  okAndNext?: ButtonProps;
+}
+
+interface DialogButtonProps {
+  cancel: ButtonProps;
+  ok: ButtonProps;
+  okAndNext: ButtonProps;
+}
+
+export interface DialogProps {
+  body: React.ReactElement;
+  buttonOverrides?: DialogButtonOverrides;
+  title?: LocaleKey;
+}
+
+interface DialogState {
+  showDialog: () => void;
+  hideDialog: () => void;
+  Modal: React.ReactElement;
+}
+
+export const useDialog = (dialogProps: DialogProps): DialogState => {
+  const { body, buttonOverrides, title } = dialogProps;
+  const [open, setOpen] = React.useState(false);
+  const t = useTranslation();
+  const showDialog = () => setOpen(true);
+  const hideDialog = () => setOpen(false);
+
+  const dialogButtonProps: DialogButtonProps = {
+    cancel: {
+      icon: <XCircleIcon />,
+      labelKey: 'button.cancel',
+      onClick: hideDialog,
+      visible: true,
+      ...buttonOverrides?.cancel,
+    },
+    ok: {
+      icon: <CheckIcon />,
+      labelKey: 'button.ok',
+      onClick: hideDialog,
+      visible: true,
+      ...buttonOverrides?.ok,
+    },
+    okAndNext: {
+      icon: <ArrowRightIcon />,
+      labelKey: 'button.ok-and-next',
+      visible: false,
+      ...buttonOverrides?.okAndNext,
+    },
+  };
+
+  const Modal = (
```

Comment:
nice - I'm changing to the JSX syntax, I think that's a great idea.
A couple of things - which I was trying to work around with the original approach.
- Using the JSX syntax requires explicitly handling the hiding of the modal in the cancel & ok buttons. What I was wanting was to have all the defaults handled. open to suggestions how to do this!
- I liked the raising of the onClose event, if the user clicks away from the modal. have implemented with an explicit handler which can be passed to the modal

Reply:
another thought, which comes back to an original idea. Would it be neater to just have a button array as a prop on the modal now, rather than explicit buttons?